pkg/controller/process.go

```diff
@@ -43,6 +43,15 @@ func (c AppGwIngressController) Process(event events.Event) error {
 		EnvVariables:         environment.GetEnv(),
 	}
 
+	// Mutate the list of Ingresses by removing ones that AGIC should not be creating configuration.
+	if cbCtx.EnvVariables.EnableBrownfieldDeployment == "true" {
+		for idx, ingress := range cbCtx.IngressList {
+			glog.V(5).Infof("Original Ingress[%d] Rules: %+v", idx, ingress.Spec.Rules)
+			brownfield.PruneIngressRules(ingress, cbCtx.ProhibitedTargets, cbCtx.ManagedTargets)
```

Comment:
can we return the filtered ingress as an `output` variable ?

Reply:
Thanks for calling this out - I like pure functions too.